python/paddle/distributed/fleet/base/distributed_strategy.py

```diff
@@ -15,10 +15,25 @@
 import paddle
 from paddle.distributed.fleet.proto import distributed_strategy_pb2
 from paddle.fluid.framework import Variable, set_flags, core
+from paddle.fluid.wrapped_decorator import wrap_decorator
 import google.protobuf.text_format
 
 __all__ = ["DistributedStrategy"]
 
+non_auto_func_called = True
+
+
+def __non_auto_func_called__(func):
+    def __impl__(*args, **kwargs):
+        global non_auto_func_called
+        non_auto_func_called = False
+        return func(*args, **kwargs)
+
+    return __impl__
+
+
+is_strict_auto = wrap_decorator(__non_auto_func_called__)
```

Comment:
This keyword is a little hard to be understood. Could we name it 'reset_auto_flag' as what it exactly does?

Reply:
this is just a check, whether a user is using strictly auto configuration. Here is a strictly auto configuration.
``` python
strategy = DistributedStrategy()
strategy.auto = True
```
Here is a case that is not a strict auto configuration.
``` python
strategy = DistributedStrategy()
strategy.amp = True
strategy.auto = True
```